common/lib/xmodule/xmodule/js/src/video/10_main.js

```diff
@@ -138,6 +140,10 @@
                     initialize(state, element);
                 }
 
+                if (autoAdvanceEnabled) {
+                    mainVideoModules.push(VideoAutoAdvanceControl);
+                }
```

Comment:
@clemente I was too specific - alternative solution would have been to move the code to 119 instead, which would also have no side effects. The concat and ternary as you've done in your most recent commit is good too. :+1: 

> I saw that there's {{state}} using {{mainVideoModules}} in it, but my code is still safe because arrays are passed by reference

While it's passed by reference, the code you had in this diff was bad practice; if `initialize` ever acts on the list of modules in the state before you append more stuff to it (if not in this version, by future changes of the codebase), we're likely to run into subtle bugs that are difficult to track down.

> But that's what I meant when I said I didn't like this approach! It's complex or ugly.

If this is more about the overall approach: I would prefer having the control not construct/destruct at all rather than the prior approach of adding a branch to the `initialization` of the control. I think this is much cleaner and less likely to cause bugs, do you agree? I'm open to discussing it further if you don't.
I also don't think it's any more complex. Could you elaborate on this?